iotedgedev/deploymentmanifest.py

```diff
@@ -151,25 +157,46 @@ def validate_deployment_template(self):
     def validate_deployment_manifest(self):
         validation_success = True
         try:
-            validation_success = self._validate_deployment_manifest_schema()
+            if self.is_layered_deployment_schema():
+                validation_success = self._validate_deployment_manifest_schema()
             validation_success &= self._validate_create_options()
         except Exception as err:
             self.output.info("Unexpected error during deployment manifest validation, skip the validation. Error:%s" % err)
 
         return validation_success
 
+    def is_layered_deployment_schema(self):
+        return "content" in self.json
```

Comment:
>"content" [](http://example.com/codeflow?start=15&length=9)

I wonder if there is a schema definition we could leverage here, similar to [https://github.com/marianan/schemastore/blob/master/src/schemas/json/azure-iot-edge-deployment-template-4.0.json](https://github.com/marianan/schemastore/blob/master/src/schemas/json/azure-iot-edge-deployment-template-4.0.json)

Reply:
I'm not sure that would work, since the schemas I have start with `content` and this requires `modulesContent` as the first prop (If  I'm reading it right